### DISCUSSION TITLE
PnP Search - Search results not displaying correct result when search filter and search vertical has been applied

### DISCUSSION THREAD

**Discussion by Rickster008** [microsoft-search/pnp-modern-search#3594] [Q&A]
**Version used:** 4.9.2

**Describe the bug:**
I am configuring the PnP search result webpart to display results from a specific library and with a certain content type applied. I have set up a search filter and Seach vertical alongside this as well. However currently when I select a search filter and  then select on the search vertical the results are not updating with the correct search results.

**To reproduce:**
Step1: Search in the search box with any keyword
Step2: Results will display in the search results, select the search verticals to ensure it has filtered based on the search criteria.
Step3: Select any search filter, click apply, Search results do not now display correct results when you select the other vertices.

Please see screenshot.

**Expected behaviour:**
Search result should display accurate results based on the search filter and search vertical applied.

**Browser:**
Edge and Chrome

**Comment by wobba** [comment]
Converted to a discussion as full repro steps are missing, and did you follow the guide at at https://microsoft-search.github.io/pnp-modern-search/usage/search-filters/#connection when connecting the filter web part and result web part both ways? And this for verticals https://microsoft-search.github.io/pnp-modern-search/scenarios/Create-a-search-page-with-verticals-within-the-same-page/.

**Reply by oliver-tbs** [reply]
Where do I find the GUID value to enter into the Result Source Id input field as per these instructions?

> **Configure second Search Result Web Part**
> The second results web part displays results from the predefined result source knowledge, configures the result source in the SharePoint admin center, and the GUID must be added to the Web Part.

**Comment by Rickster008** [comment]
@wobba Thanks for sending through the articles, really useful.
